interp/run.go

```diff
@@ -1578,6 +1577,7 @@ func _range(n *node) {
 	// Init sequence
 	next := n.exec
 	n.child[0].exec = func(f *frame) bltn {
+		a = value(f)
```

Comment:
we should have a frame location (say index1) to store the array length:

```go
v1 := f.data[index1]
v1.SetInt(value(f).Len())
```

Reply:
As pointed out, this will still fail to match the compiled code's behaviour. Making a new value that is outside the scope of the interpreted code is far more aligned with the compiled code case where the range expression is not available to the Go program using anything except unsafe access.

In both the slice and array cases the value of the range expression is copied (it is dangerous to think of arrays and slices as arrays since this leads to misconceptions of shared behaviours). In the case of a slice, we have approximately what you are suggesting while in the array case you get an entire copy of the array. Now you could differentiate and copy for the array and keep the length for the slice, but we already get that behaviour with the change here.

Reply:
Agree. No need to store just the index as we have to get access to the entire array copy anyway